docs/site/tutorials/protocol_oriented_generics.ipynb

```diff
@@ -0,0 +1,1141 @@
+{
```

Comment:
<p>1) It would be good to define what's a protocol-oriented way, so that the reader understands what exactly we find desirable about this way of developing software.</p><p><br></p><p>2) Also, how does one quantify abstraction? Just like with "very expressive", "high level of abstraction" lacks established quantification criteria and therefore may invite controversy, e.g.: "In my favorite language, it is impossible to do this. Does this now mean that my favorite language has a low level of abstraction (i.e. is "bad")?"</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/swift/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>